src/prefect/tasks/gsheets/gsheets.py

```diff
@@ -0,0 +1,160 @@
+from prefect.utilities.tasks import defaults_from_attrs
+import gspread
+from typing import Dict, List, Tuple, Union, Any, TypedDict, Callable
+import prefect
+from prefect import task, Flow, Parameter, Task
+from prefect.tasks.secrets.base import SecretBase
+from prefect.tasks.secrets import PrefectSecret
+import pathlib
+from pathlib import Path
+
+class GsheetUpdates(TypedDict):
+    """A Typed Dict to describe one of the dictionary elements returned after writing Rows"""
+    spreadsheetId: str
+    updatedRange: str
+    updatedRows: int
+    updatedColumns: int
+    updatedCells: int
+
+class GsheetResponse(TypedDict):
+    """A Typed Dict to describe what's returned after writing Rows"""
+    spreadsheetId: str
+    tableRange: str
+    updates: GsheetUpdates
+        
+class AuthenticateGsheets(SecretBase):
+    """
+    A Secret Task for creating the Google Client object needed to perform operations
+    with `gspread`.  Presumes that you'll be interacting with Sheets via a Service
+    Account: https://gspread.readthedocs.io/en/latest/oauth2.html#for-bots-using-service-account
+    Args:
+        - credentials_filename (Union[str, pathlib.Path]): Location of credentials file
+    """
+    def __init__(self, credentials_filename: Union[str, pathlib.Path], **kwargs: Any):
+        self.credentials_filename = credentials_filename
+        super().__init__(**kwargs)
+
+    def run(self) -> gspread.client.Client:
+        return gspread.service_account(filename=self.credentials_filename)

```

Comment:
Hmmm this won't work in a lot of situations because clients are not serializable objects.  If you want to separate out this logic into a utility function that's perfectly valid but it shouldn't be structured as a task

Reply:
K, I replaced it with direct calls to the Client Creator.  Hopefully this won't be cached or anything?  Someone told me a Secret Task was the best way to make sure that didn't happen.

Reply:
So there are a few things here: the only objects that are at risk of being cached are explicit task inputs and outputs.  In the special case that one of your task's upstream inputs is actually a `Secret` task, Prefect will ensure that that particular input is not cached / stored and is instead always recomputed.  

In the original code you had, you weren't using the Auth task as a `Task`, but were instead calling its `run` method directly within another task which is equivalent to the new code that you have here but less explicit.

Hope that explanation helps!

Reply:
Awesome, thanks!  Cool learning Prefect's internals :)